src/main/java/com/amazon/opendistroforelasticsearch/security/transport/OpenDistroSecurityRequestHandler.java

```diff
@@ -126,12 +127,20 @@ protected void messageReceivedDecorate(final T request, final TransportRequestHa
               && !transportChannel.getChannelType().equals("PerformanceAnalyzerTransportChannelType")) {
                throw new RuntimeException("Unknown channel type "+transportChannel.getChannelType());
            }
+           String channelType = transportChannel.getChannelType();
 
-           getThreadContext().putTransient(ConfigConstants.OPENDISTRO_SECURITY_CHANNEL_TYPE, transportChannel.getChannelType());
+           if(transportChannel.getChannelType().equals("PerformanceAnalyzerTransportChannelType")) {
```

Comment:
we can reuse channelType string here.

if(channelType.equals("PerformanceAnalyzerTransportChannelType")) {